physicsnemo/models/domino/model.py

```diff
@@ -55,55 +51,80 @@ def calculate_pos_encoding(nx, d=8):
     return vec
 
 
-def scale_sdf(sdf):
-    """Function to scale SDF"""
-    return sdf / (0.4 + abs(sdf))
-
-
-def calculate_gradient(sdf):
-    """Function to calculate the gradients of SDF"""
-    m, n, o = sdf.shape[2], sdf.shape[3], sdf.shape[4]
-    sdf_x = sdf[:, :, 2:m, :, :] - sdf[:, :, 0 : m - 2, :, :]
-    sdf_y = sdf[:, :, :, 2:n, :] - sdf[:, :, :, 0 : n - 2, :]
-    sdf_z = sdf[:, :, :, :, 2:o] - sdf[:, :, :, :, 0 : o - 2]
-
-    sdf_x = F.pad(input=sdf_x, pad=(0, 0, 0, 0, 0, 1), mode="constant", value=0.0)
-    sdf_x = F.pad(input=sdf_x, pad=(0, 0, 0, 0, 1, 0), mode="constant", value=0.0)
-    sdf_y = F.pad(input=sdf_y, pad=(0, 0, 0, 1, 0, 0), mode="constant", value=0.0)
-    sdf_y = F.pad(input=sdf_y, pad=(0, 0, 1, 0, 0, 0), mode="constant", value=0.0)
-    sdf_z = F.pad(input=sdf_z, pad=(0, 1, 0, 0, 0, 0), mode="constant", value=0.0)
-    sdf_z = F.pad(input=sdf_z, pad=(1, 0, 0, 0, 0, 0), mode="constant", value=0.0)
+def scale_sdf(sdf: torch.Tensor) -> torch.Tensor:
+    """
+    Scale a signed distance function (SDF) to emphasize surface regions.
 
-    return sdf_x, sdf_y, sdf_z
+    This function applies a non-linear scaling to the SDF values that compresses
+    the range while preserving the sign, effectively giving more weight to points
+    near surfaces where |SDF| is small.
 
+    Args:
+        sdf: Tensor containing signed distance function values
 
-def binarize_sdf(sdf):
-    """Function to calculate the binarize the SDF"""
-    sdf = torch.where(sdf >= 0, 0.0, 1.0)
-    return sdf
+    Returns:
+        Tensor with scaled SDF values in range [-1, 1]
+    """
+    return sdf / (0.4 + torch.abs(sdf))
 
 
 class BQWarp(nn.Module):
-    """Warp based ball-query layer"""
+    """
+    Warp-based ball-query layer for finding neighboring points within a specified radius.
+
+    This layer uses an accelerated ball query implementation to efficiently find points
+    within a specified radius of query points.
+    """
 
     def __init__(
         self,
-        input_features,
-        grid_resolution=[256, 96, 64],
-        radius=0.25,
-        neighbors_in_radius=10,
+        input_features: int,
```

Comment:
Is input_features even used?

Reply:
Fixed in b572a2c279210ec3e25e1af902cee9384108773c